compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -691,7 +691,9 @@ private void addToFunctionMap(BLangFunction funcNode, SymbolEnv symbolEnv, int p
     @Override
     public void visit(BLangBlockStmt blockNode) {
         SymbolEnv blockEnv = SymbolEnv.createBlockEnv(blockNode, env);
-        blockClosureMapCount++;
+         if (!blockNode.internal) {
+            blockClosureMapCount++;
+         }
```

Comment:
Is there any specific reason for doing this check? I don't think it's necessary to handle it as a special case by verifying whether that block-stmt is an internal node or not.

Reply:
We no longer need to create a blockClosureMap for an internal block that we created. In the case of a internal block node we created the closure variables would be attached to the parent of the created block.